### DISCUSSION TITLE
[Question]:Attache excel file is too slow more than Chatgpt

### DISCUSSION THREAD

**Discussion by angpao** [danny-avila/LibreChat#4453] [Q&A]
### What is your question?

When an Excel file is uploaded, the system needs to read the Excel file as text, perform chunking, and send the chunks for embedding in the LLM model. The resulting data will then be inserted into pgvector. Each chunk will be inserted into pgvector individually. I’m not sure if you can make the system perform a bulk insert all at once, which might improve the speed. Can you do that?

### More Details

When upload file in chatgpt is very fast but Librechat is too slow. How to solve?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It’s not possible to do a bulk insert all at once because embeddings need to be generated, which is rate limited. I’ve experimented with this and even when all chunks are processed at once, the external embeddings service (OpenAI), will seem to be delayed the same amount of time for each chunk.

Furthermore, I’m sure that ChatGPT is not doing a RAG method for excel files at all, it’s possible they are doing a data manipulation, text-to-sql like process that is in the works on our end.

**Comment by angpao** [comment]
Do you mean the issue comes from the rate limit and RAG method (Chatgpt no use embedding model but text to sql ,right? Can you show more in details)? If I want to solve the rate limit issue of Azure OpenAI embeddings, what should I do? And as for the RAG method, when will it be available once you’ve further developed it?
And can LibreChat use another embedding model instead? How can it be done? (Currently using Azure OpenAI Service)
